up-l1/mqtt_5.adoc

```diff
@@ -24,165 +24,215 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
 
 For more information, please refer to https://mqtt.org/
 
-This document will discuss the uTransport implementation on MQTT 5.
+This document will discuss the uTransport implementation on MQTT 5 for two well known use cases:
+1. *Device-2-Device (D2D) Communication:* This is when devices (through the streamer) was to send messages to other devices through a cloud gateway (broker). 
+2. *uE-2-uE Communication:* When MQTT is used as a local software bus for uEs to talk to each other through a local broker.
 
-== Specifications
+The reason fo the different configurations is that cloud MQTT brokers often have limitations on the number of topics that can be subscribed to as well as the topic segment lengths so we need to ensure that message routing is as efficient as possible.
 
-=== UAttributes Mapping
+== MQTT5 Header
 
-MQTT 5 supports custom header fields, and we leverage this to map the UAttributes values into the MQTT header.
+MQTT 5 supports custom header fields, and we leverage this to map the `UAttributes` values into the `UserProperty` MQTT header in string format.
 
-[cols="1,1,1,1"]
+[.specitem,oft-sid="req~up-transport-mqtt5-attributes~1",oft-needs="impl,utest"]
+--
+* `UserProperty` Keys *MUST* map to the UAttributes protobuf field numbers.
+--
```

Comment:
FMPOV there is no need for this as you are explicitly listing all attributes in the following table, along with the keys to use.

However, what we probably should define is that only non-empty attribute values should be mapped to MQTT header fields.